pulser-core/pulser/sequence/sequence.py

```diff
@@ -437,8 +475,102 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
         if self._slm_mask_targets:
             raise ValueError("SLM mask can be configured only once.")
 
-        # If checks have passed, set the SLM mask targets
         self._slm_mask_targets = targets
+        self._slm_mask_dmm = dmm_id
+        if self._in_xy and dmm_id not in self._device.dmm_channels:
+            raise ValueError("No DMM %s in the device." % dmm_id)
+        elif not self._in_xy:
+            ntargets = len(targets)
+            detuning_map = self.register.define_detuning_map(
+                {
+                    qubit: (1 / ntargets if qubit in targets else 0)
+                    for qubit in self.register.qubit_ids
+                }
+            )
+            self.config_detuning_map(detuning_map, dmm_id)
+            for key in reversed(self.declared_channels.keys()):
+                if dmm_id in key:
+                    self._slm_mask_dmm = key
+                    break
+
+    def insert_slm_mask(self) -> None:
```

Comment:
Should this be something the user can call? I think once `config_slm_mask` is called, we should handle the insertion internally